### PR TITLE
[2.7] Other small edits due to 2.7 (exkl. UI guide)

### PR DESCRIPTION
**Pull request by TJaniF**
Ready to be merged :)
_____
- adds a description of the `fail_stop` DAG parameter
- updates UI screenshots in the getting started guide (there are many UI screenshots that need an update now, anything that shows the old graph view for example 😅 but I thought having the getting started tutorial up to date is the most important one)
- remove (hopefully all?) references to testing connections or link to how to enable that feature, also add notes to the testing connections sections 
- Added notes about Celery and Kubernetes executors having been moved into their providers (and yes, updating the whole executor guide is on my backlog already 😅 )
- The `Trigger DAG w/config` option is gone on manual trigger. It is all `Trigger DAG` now and will open the config view when Params are defined, otherwise just run the DAG without config.
- adding a small note about the ExternalBranchPythonOperator

Some small changes like a scheduler config default value, how to access task logs etc.

### CODE REVIEW

File: learn/airflow-executors-explained.md [hunk 1]
```diff
@@ -100,7 +100,7 @@ At its core, the Celery executor is built for horizontal scaling.
 
 If a worker node is ever down or goes offline, the Celery executor quickly adapts and is able to assign that allocated task or tasks to another worker.
 
-If you're running native Airflow, adopting a Celery executor means you'll have to set up an underlying database to support it (RabbitMQ/Redis).
+If you're running native Airflow, adopting a Celery executor means you'll have to set up an underlying database to support it (RabbitMQ/Redis). The Celery executor is available as part of the [Celery provider](https://registry.astronomer.io/providers/apache-airflow-providers-celery/versions/latest).
```
Comment (kentdanas) [inline learn/airflow-executors-explained.md:103]:
```suggestion
If you're running native Airflow, adopting a Celery executor means you'll have to set up an underlying database to support it (RabbitMQ/Redis). The Celery executor is available as part of the [Celery provider](https://registry.astronomer.io/providers/apache-airflow-providers-celery/versions/latest) and must be installed separately from core Airflow.
```
Let me know what you think about adding this. Just to reinforce that it won't come prebaked.

File: learn/airflow-executors-explained.md [hunk 2]
```diff
@@ -139,6 +139,8 @@ This means a few things:
     - Service accounts
     - Airflow image
 
+The Kubernetes executor is available as part of the [CNCF Kubernetes provider](https://registry.astronomer.io/providers/apache-airflow-providers-cncf-kubernetes/versions/latest).
```
Comment (kentdanas) [inline learn/airflow-executors-explained.md:142]:
```suggestion
The Kubernetes executor is available as part of the [CNCF Kubernetes provider](https://registry.astronomer.io/providers/apache-airflow-providers-cncf-kubernetes/versions/latest) and must be installed separately from core Airflow.
```

**Review by kentdanas — APPROVED** [review]
Looks great!

File: learn/airflow-scaling-workers.md [hunk 3]
```diff
@@ -87,7 +87,7 @@ Scheduler settings control how the scheduler parses DAG files and creates DAG ru
 
 - `max_dagruns_to_create_per_loop`: The maximum number of DAGs to create DAG runs for per scheduler loop. Decrease the value to free resources for scheduling tasks. The default value is 10. 
 
-- `max_tis_per_query`: Changes the batch size of queries to the metastore in the main scheduling loop. A higher value allows more `tis` to be processed per query, but your query may become too complex and cause performance issues. The default value is 512 queries. 
+- `max_tis_per_query`: Changes the batch size of queries to the metastore in the main scheduling loop. A higher value allows more `tis` to be processed per query, but your query may become too complex and cause performance issues. The default value is 16 queries in Airflow 2.7+. Note that the `scheduler.max_tis_per_query` value needs to be lower than the `core.parallelism` value.
```
Comment (jwitz) [inline learn/airflow-scaling-workers.md:90]:
```suggestion
- `max_tis_per_query`: Changes the batch size of queries to the metastore in the main scheduling loop. A higher value allows more `tis` to be processed per query, but your query may become too complex and cause performance issues. The default value is 16 queries. Note that the `scheduler.max_tis_per_query` value needs to be lower than the `core.parallelism` value.
```
This seems like one of those that we can assume as the new default for Airflow

File: learn/debugging-dags.md [hunk 4]
```diff
@@ -229,6 +229,12 @@ The following are some debugging steps you can try:
 
     ![Test Connections](/img/guides/test_connections_2.png)
 
+    :::note
+
+    In Airflow 2.7+ testing connections is disabled by default. You can enable connection testing by setting the [`test_connection` core config](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#test-connection) to `Enabled` by defining the environment variable `AIRFLOW__CORE__TEST_CONNECTION=Enabled` in your Airflow environment. Astronomer recommends not to enable this feature until you made sure that only highly trusted UI/API users have "edit connection" permissions.
```
Comment (jwitz) [inline learn/debugging-dags.md:234]:
```suggestion
    Testing connections is disabled by default in Airflow 2.7+. You can enable connection testing by defining the environment variable `AIRFLOW__CORE__TEST_CONNECTION=Enabled` in your Airflow environment. Astronomer recommends not enabling this feature until you are sure that only highly trusted UI/API users have "edit connection" permissions.
```

**Review by jwitz — APPROVED** [review]

**Comment by TJaniF** [comment]
ready :)

**Pull request merged**